src/logger/BaseLogger_test.go

```diff
@@ -0,0 +1,228 @@
+	//replacement of the java "startswith" assertion
+	//won't work rn since version is a dummy string
+	assert.Equal(t, strings.Index(version, "2.0."), 0)
+	assert.NotContains(t, version, "\\")
+	assert.NotContains(t, version, "\"")
+	assert.NotContains(t, version, "'")
+	assert.Equal(t, baseLogger.Version, version)
+}
+
+func TestPerformsEnablingWhenExpected(t *testing.T) {
+	helper := NewTestHelper()
+	logger := NewBaseLogger(helper.mockAgent, helper.demoURL, false, nil)
+	assert.True(t, logger.Enableable)
+	assert.False(t, logger.Enabled)
+	assert.Equal(t, logger.Url, helper.demoURL)
+	logger.Enable()
+	assert.True(t, logger.Enabled)
+
+	queue := []string{}
+	logger = NewBaseLoggerAgentQueueEnabled(helper.mockAgent, queue, false)
+	assert.True(t, logger.Enableable)
+	assert.False(t, logger.Enabled)
+	assert.Nil(t, logger.Url)
+	logger.Enable()
+	logger.Disable()
+	logger.Enable()
+	assert.True(t, logger.Enabled)
+}
+
+//needs some more stuff in the baselogger class for this to compile
+func TestSkipsEnablingForInvalidUrls(t *testing.T) {
+	helper := NewTestHelper()
+	for _, url := range helper.mockURLSinvalid {
+		logger := NewBaseLogger(MOCK_AGENT, url, false, nil)
+		assert.False(t, logger.Enableable)
+		assert.False(t, logger.Enabled)
+		assert.Nil(t, logger.Url)
+		logger.Enable()
+		logger.Disable()
+		logger.Enable()
+		assert.False(t, logger.Enabled)
+	}
+}
+
+func TestSkipsEnablingForNullUrls(t *testing.T) {
+	url := ""
+	helper := NewTestHelper()
+	logger := NewBaseLoggerAgentUrl(helper.mockAgent, url)
+	assert.False(t, logger.Enableable)
+	assert.False(t, logger.Enabled)
+	assert.Nil(t, logger.Url)
+	logger.Enable()
+	assert.False(t, logger.Enabled)
+}
+
+func TestSubmitsToDemoUrl(t *testing.T) {
+	helper := NewTestHelper()
+	logger := NewBaseLoggerAgentUrl(helper.mockAgent, helper.demoURL)
+	message := [][]string{}
+	message = append(message, []string{"agent", logger.Agent})
+	message = append(message, []string{"version", logger.Version})
+	message = append(message, []string{"now", string(helper.mockNow)})
+	message = append(message, []string{"protocol", "https"})
+	//TODO: toimplement stringify method
+	msg := "Json.stringify(message)"
+	//TODO: implement parsable method
+	assert.True(t, "parsable(msg)")
+	logger.Submit(msg)
+	assert.Equal(t, logger.SubmitFailures, 0)
+	assert.Equal(t, logger.SubmitSuccesses, 1)
+}
+
+func TestSubmitsToDemoUrlViaHttp(t *testing.T) {
+	helper := NewTestHelper()
+	logger := NewBaseLoggerAgentUrl(helper.mockAgent, strings.Replace(helper.demoURL, "https://", "http://", 1))
+	assert.Equal(t, strings.Index(logger.Url, "http://"), 0)
+	message := [][]string{}
+	message = append(message, []string{"agent", logger.Agent})
+	message = append(message, []string{"version", logger.Version})
+	message = append(message, []string{"now", string(helper.mockNow)})
+	message = append(message, []string{"protocol", "http"})
+	//TODO: toimplement stringify method
+	msg := "Json.stringify(message)"
```

Comment:
Tim mentioned above but don't need the stringfy. Double check with Tim

Reply:
yeah this is fine for now